model/post.go

```diff
@@ -90,6 +92,10 @@ type Post struct {
 	// Transient data populated before sending a post to the client
 	ReplyCount int64         `json:"reply_count" db:"-"`
 	Metadata   *PostMetadata `json:"metadata,omitempty" db:"-"`
+
+	// Unexported sync.RWMutex used to prevent data race conditions.
+	// It's currently only used to guard Post.Props.
+	mu sync.RWMutex `db:"-"`
```

Comment:
It's idiomatic to keep the mutex on top of the field it is protecting, so that it becomes more readable. Let's do that here.

Reply:
@agnivade the idea was to keep it separated since it's the only unexported field in the struct, same as we do in other cases. That said, I am fine with moving it up if you feel it makes the code more clear.

Reply:
I think it does. Having it on top of Props, and naming it propsMu makes it dead simple. Otherwise, I have to read the comment and go up and see what the Props field is.

Keeping unexported fields in a separate group is totally fine. But in cases of mutexes guarding a field, it is better to keep them close together.

Some examples from the Go codebase:

```
cmd/compile/internal/gc/go.go:  funcsymsmu sync.Mutex // protects funcsyms and associated package lookups (see func funcsym)
cmd/compile/internal/gc/go.go-  funcsyms   []*types.Sym
cmd/compile/internal/gc/go.go-)
--
cmd/compile/internal/gc/reflect.go:     signatmu    sync.Mutex // protects signatset and signatslice
cmd/compile/internal/gc/reflect.go-     signatset   = make(map[*types.Type]struct{})
cmd/compile/internal/gc/reflect.go-     signatslice []*types.Type
--
cmd/compile/internal/gc/reflect.go:     typepkgmu sync.Mutex // protects typepkg lookups
cmd/compile/internal/gc/reflect.go-     typepkg   = types.NewPkg("type", "type")
cmd/compile/internal/gc/reflect.go-)
--
cmd/compile/internal/gc/subr.go:        largeStackFramesMu sync.Mutex // protects largeStackFrames
cmd/compile/internal/gc/subr.go-        largeStackFrames   []largeStack
cmd/compile/internal/gc/subr.go-)
--
cmd/compile/internal/logopt/log_opts.go:var mu = sync.Mutex{} // mu protects loggedOpts.
cmd/compile/internal/logopt/log_opts.go-
cmd/compile/internal/logopt/log_opts.go-func LogOpt(pos src.XPos, what, pass, fname string, args ...interface{}) {
--
cmd/compile/internal/types/pkg.go:      internedStringsmu sync.Mutex // protects internedStrings
cmd/compile/internal/types/pkg.go-      internedStrings   = map[string]string{}
cmd/compile/internal/types/pkg.go-)
```